WooCommerce/Classes/ViewRelated/Dashboard/Settings/SettingsViewController.swift

```diff
@@ -38,6 +38,11 @@ final class SettingsViewController: UIViewController {
         return urlAsString?.hostname() ?? String()
     }
 
+    private var siteID: Int64 {
+        let siteID = ServiceLocator.stores.sessionManager.defaultStoreID
+        return siteID ?? 0
+    }
```

Comment:
Since it's used only in a single place, what do you think about moving it there? Also fallback to 0 will break things on CouponManagementViewController, maybe add guard with error message instead?

Reply:
I've added a pop up error message if there is no store ID, with an action to let the user select the store. Was this what you meant for the error message? I think it's probably better than a `fatalError` in this instance, since it should be user-recoverable really.
![coupon-list-opening-error](https://user-images.githubusercontent.com/80399824/117415165-e833a100-af0f-11eb-8ee3-d8d5c8d95b13.png)